astropy/coordinates/attributes.py

```diff
@@ -485,6 +498,9 @@ def convert_input(self, value):
             If the input is not valid for this attribute.
         """
 
+        if value is None:
```

Comment:
I don't think this is needed - no tests fail if I remove it (see general comment)